packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2015,6 +2015,7 @@ class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetec
     if (!_isDoubleTap) {
       widget.onSingleTapUp?.call(details);
       _lastTapOffset = details.globalPosition;
+      _doubleTapTimer?.cancel();
```

Comment:
The change here doesn't seem to be related?

Reply:
Without canceling the old timer, the `A Focused text-field will not lose focus when clicking on its decoration` test in the `text_field_focus_test.dart` will be failed with this error:

> A Timer is still pending even after the widget tree was disposed

Due to reducing the `_kTransitionDuration`, something happens too quickly.